### ISSUE TITLE
Undesired rewriting of import statements (pre-commit loop)

### ISSUE THREAD

**Issue by benjaoming** [adamchainz/django-upgrade#250]
### Python Version

3.8

### Django Version

2.2

### Package Version

main

### Description

I need some help triaging this behavior, as I'm not familiar with the patterns that django-upgrade uses to detect and alter import statements.

While running django-upgrade on a project, I have the following import statement generated:

```python
from django.urls import include, path, re_path
```

However, if my project is using `reorder_python_imports`, it will split up this line into 3 import statements:

```python
from django.urls import include
from django.urls import path
from django.urls import re_path
```

Now, if I run the `django-upgrade` hook again, the former line reappears like this:

```python
from django.urls import include
from django.urls import path
from django.urls import include, path, re_path
```

You can add the latest version of `reorder_python_imports` with this configuration:

```
-   repo: https://github.com/asottile/reorder_python_imports
    rev: v3.8.3
    hooks:
    -   id: reorder-python-imports
        language_version: python3

```

Are you able to immediately grasp the issue @adamchainz or does it need to be isolated into a test project / test case?

### The correct behavior?

I think that what I would expect from django-upgrade is that it generates whatever import statements that it needs, but that subsequent linting from `black`, `flake8` or `reorder_python_imports` is left in place and doesn't result in any new changes.

**Comment by adamchainz** [comment]
The import should only be added if something else in the file needs it - can you see any URL's that are being modified as well?

**Comment by benjaoming** [comment]
Nothing else is being modified.

I believe this happens:

1) I ran `precommit run django-upgrade --all-files` and initially, things were modified such that the new import statements made sense together with URL pattern changes
2) Upon "git commit", I see that subsequent pre-commit hooks make additional changes
3) I commit the changes made by pre-commit but since django-upgrade is ALSO installed as a pre-commit, I now see that it has made new import statement changes as described above (but nothing else got changed)

**Comment by adamchainz** [comment]
Thanks, replicated with an extra test in `test_django_urls.py`:

```python
def test_two_imported_used():
    check_noop(
        """\
        from django.urls import path
        from django.urls import re_path

        path('whatever')
        re_path('whatever')
        """,
        settings,
    )
```

The problem is that the call-rewriter is registering `re_path` as "used" when it encounters it (even though it doesn't rewrite it to `path`), which makes the import-rewriter rewrite it.

I think the import rewriting code here could do with some improvements to support all import styles.

**Comment by sshishov** [comment]
The same issue is happening for us.

We are also using one line per import and the library is rewriting the import and then isort rewrite it back.

Means we cannot use this `pre-commit` until the issue is fixed.

**Comment by adamchainz** [comment]
Understood, I started lookign a fix in #255, treating it as a release blocker at least.

**Comment by sshishov** [comment]
Thank you for the update @adamchainz 👍
